File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task22_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
#include <vector>
#include <iostream>

std::vector<int> filterList(std::vector<Object> list) {
    std::vector<int> result;
    for (auto obj : list) {
        if (typeid(obj) == typeid(int)) {
            result.push_back(obj);
        }
    }
    return result;
}

File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task22_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import java.util.ArrayList;
import java.util.List;

public class Kata {
    public static List<Object> filterList(List<Object> list) {
        List<Object> newList = new ArrayList<>();
        for (Object obj : list) {
            if (obj instanceof Integer) {
                newList.add(obj);
            }
        }
        return newList;
    }
}